src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java

```diff
@@ -180,19 +208,86 @@ public void close() {
 
   @Override
   public List<String> getLiveNodes() throws ReaperException {
-    return null; // TODO: implement me.
+    try {
+      List<String> liveNodes = new ArrayList<>();
+      EndpointStates endpoints = apiClient.getEndpointStates();
+      for (Map<String, String> states : endpoints.getEntity()) {
+        if (states.containsKey("IS_ALIVE") && Boolean.parseBoolean(states.get("IS_ALIVE"))) {
+          liveNodes.add(states.get("ENDPOINT_IP"));
+        }
+      }
+      return liveNodes;
+    } catch (ApiException ae) {
+      LOG.error("Failed to fetch live nodes", ae);
+      throw new ReaperException(ae);
+    }
   }
 
-  public void clearSnapshot(String var1, String... var2) throws IOException {
-    // TODO: implement me.
+  public void clearSnapshot(String snapshotName, String... keyspaces) throws IOException {
+    try {
+      apiClient.clearSnapshots(Arrays.asList(snapshotName), Arrays.asList(keyspaces));
+    } catch (ApiException ae) {
+      LOG.error("Failed to clear snapshots", ae);
+      throw new IOException(ae);
+    }
   }
 
   public List<Snapshot> listSnapshots() throws UnsupportedOperationException {
-    return new ArrayList<Snapshot>();
+    try {
+      SnapshotDetails snapshotDetails = apiClient.getSnapshotDetails(null, null);
+      return convert(snapshotDetails);
+    } catch (ApiException ae) {
+      LOG.error("Failed to retrieve Snapshot details", ae);
+      throw new UnsupportedOperationException(ae);
+    }
   }
 
-  public void takeSnapshot(String var1, String... var2) throws IOException {
-    // TODO: implement me.
+  @VisibleForTesting
+  List<Snapshot> convert(SnapshotDetails snapshotDetails) {
```

Comment:
Issue: `convertSnapshot()` would be more appropriate here.

Reply:
I'm going with `convertSnapshots()` as it is likely more than one in the SnapshotDetails object that it will convert into a List.